components/stripe/actions/confirm-payment-intent/confirm-payment-intent.mjs

```diff
@@ -25,47 +25,310 @@ export default {
+      optional: true,
+    },
+    returnUrl: {
+      type: "string",
+      label: "Return URL",
+      description: "The URL to redirect your customer back to after they authenticate or cancel their payment on the payment method's app or site. If you'd prefer to redirect to a mobile application, you can alternatively supply an application URI scheme. This parameter is only used for cards and other redirect-based payment methods.",
+      optional: true,
+    },
+    useStripeSdk: {
+      type: "boolean",
+      label: "Use Stripe SDK",
+      description: "Set to true when confirming server-side and using Stripe.js, iOS, or Android client-side SDKs to handle the next actions.",
+      optional: true,
     },
   },
   async run({ $ }) {
-    const params = pick(this, [
-      "payment_method",
-      "receipt_email",
-      "setup_future_usage",
-    ]);
-    const resp = await this.app.sdk().paymentIntents.confirm(this.id, {
-      ...params,
-      ...this.advanced,
+    const {
+      app,
+      id,
+      paymentMethod,
+      receiptEmail,
+      setupFutureUsage,
+      shippingAddressCity,
+      shippingAddressCountry,
+      shippingAddressLine1,
+      shippingAddressLine2,
+      shippingAddressPostalCode,
+      shippingAddressState,
+      shippingName,
+      shippingCarrier,
+      shippingPhone,
+      shippingTrackingNumber,
+      captureMethod,
+      confirmationToken,
+      errorOnRequiresAction,
+      mandate,
+      mandateDataCustomerAcceptanceType,
+      mandateDataCustomerAcceptanceAcceptedAt,
+      mandateDataCustomerAcceptanceOffline,
+      mandateDataCustomerAcceptanceOnlineIpAddress,
+      mandateDataCustomerAcceptanceOnlineUserAgent,
+      offSession,
+      paymentMethodData,
+      paymentMethodOptions,
+      paymentMethodTypes,
+      radarOptionsSession,
+      returnUrl,
+      useStripeSdk,
+    } = this;
+
+    const resp = await app.sdk().paymentIntents.confirm(id, {
+      payment_method: paymentMethod,
+      receipt_email: receiptEmail,
+      setup_future_usage: setupFutureUsage,
+      ...(
+        shippingAddressCity
+        || shippingAddressCountry
+        || shippingAddressLine1
+        || shippingAddressLine2
+        || shippingAddressPostalCode
+        || shippingAddressState
+        || shippingName
+        || shippingCarrier
+        || shippingPhone
+        || shippingTrackingNumber
+          ? {
+            shipping: {
+              address: {
+                city: shippingAddressCity,
+                country: shippingAddressCountry,
+                line1: shippingAddressLine1,
+                line2: shippingAddressLine2,
+                postal_code: shippingAddressPostalCode,
+                state: shippingAddressState,
+              },
+              name: shippingName,
+              carrier: shippingCarrier,
+              phone: shippingPhone,
+              tracking_number: shippingTrackingNumber,
+            },
+          }
+          : {}
```

Comment:
Do you think we could move the logic for building these very large parameters to a method in the component? Just to keep the request structure a bit cleaner for maintainability